### ISSUE TITLE
Add start_from_epoch parameter to EarlyStopping 

### ISSUE THREAD

**Issue by klaramarijan** [Lightning-AI/lightning#16881]
### Description & Motivation

It would be great if EarlyStopping callback could have an additional parameter, start_from_epoch/star_from_iteration.

The parameter would represent the number of epochs/iterations to wait before starting to monitor improvement. 
This would allow a warm-up period (in which no improvement is expected) and thus training will not be stopped in this period, and the patience can still be as short as the user would want it to be.

Keras has this feature: https://keras.io/api/callbacks/early_stopping/

**Comment by samgelman** [comment]
I think this would be a useful feature. I actually implemented my own workaround for it previously:

```
class DelayedStartEarlyStopping(EarlyStopping):
    def __init__(self, start_epoch, *args, **kwargs):
        super().__init__(*args, **kwargs)
        # set start_epoch to None or 0 for no delay
        self.start_epoch = start_epoch

    def on_train_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
        if (self.start_epoch is not None) and (trainer.current_epoch < self.start_epoch):
            return
        super().on_train_epoch_end(trainer, pl_module)

    def on_validation_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
        if (self.start_epoch is not None) and (trainer.current_epoch < self.start_epoch):
            return
        super().on_validation_end(trainer, pl_module)
```

**Comment by aditya0by0** [comment]
Hi @awaelchli, would like to work on this, if its already **not** in wip.
